XiEditor/HoverViewController.swift

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class HoverView: NSTextView {
+    // Lets AppKit do all the work when setting up a new text view
+    override init(frame frameRect: NSRect, textContainer container: NSTextContainer?) {
+        let textView = NSTextView(frame: .zero)
+        super.init(frame: frameRect, textContainer: textView.textContainer)
+    }
+
+    init(content: String) {
+        super.init(frame: .zero)
+        self.string = content
+        self.isEditable = false
+        self.textContainerInset = NSSize(width: 10, height: 10)
+        self.font = NSFont.systemFont(ofSize: 11)
+        self.textColor = NSColor.textColor
+        self.needsLayout = true
+        self.isVerticallyResizable = true
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class HoverViewController: NSViewController {
+    lazy var scrollView: NSScrollView = {
+        let scrollView = NSScrollView(frame: NSRect(x: 0, y: 0, width: hoverPopoverWidth, height: 0))
+        scrollView.borderType = .noBorder
+        scrollView.hasVerticalScroller = true
+        scrollView.autoresizingMask = [.height]
+        return scrollView
+    }()
+    var resultContent: String
+    var hoverView: HoverView
+    let hoverPopoverWidth: CGFloat = 500 // XCode size for quick help popovers
```

Comment:
I think we should probably have a minimum and a maximum width, and then we resize ourself to the preferred width of our text, within those constraints.

Reply:
I tried this, and there is a whole host of unintended behaviors when I tried to get the view to size itself to fit all the text perfectly. Through much trial and error I and lots of reading I realized many macOS popovers (not just in XCode) works best when they have a fixed width or fixed height, because it's much easier to calculate how much space text would take in one direction. 

Also, I think it looks a bit more aesthetically more pleasing this way. :)